ci-operator/step-registry/assisted/nutanix/postinstall/assisted-nutanix-postinstall-commands.sh

```diff
@@ -91,73 +105,92 @@ cat <<EOF | oc create -f -
-      echo "CatalogSource is now READY."
-      if oc get packagemanifests nutanixcsioperator &> /dev/null; then
-        echo "Package Manifests nutanixcsioperator is now READY."
-        break
-      fi
-      echo "Waiting for nutanixcsioperator package manifests to become READY ..."
-    fi
+start_time=$(date +%s)
+while true; do
+  current_time=$(date +%s)
+  elapsed_time=$((current_time - start_time))
 
-    if [[ ${elapsed_time} -ge ${resource_timeout_seconds} ]]; then
-      echo "Timeout: Nutanix CSI CatalogSource did not become READY within ${resource_timeout_seconds} seconds."
-      exit 1
+  if [[ $(oc get catalogsource nutanix-csi-operator-beta -n openshift-marketplace -o 'jsonpath={..status.connectionState.lastObservedState}') == "READY" ]]; then
+    echo "CatalogSource is now READY."
+    if oc get packagemanifests nutanixcsioperator &> /dev/null; then
+      echo "Package Manifests nutanixcsioperator is now READY."
+      break
     fi
+    echo "Waiting for nutanixcsioperator package manifests to become READY ..."
+  fi
 
-    echo "Waiting for CatalogSource to be READY..."
-    sleep 5s
-  done
-fi
+  if [[ ${elapsed_time} -ge ${resource_timeout_seconds} ]]; then
+    echo "Timeout: Nutanix CSI CatalogSource did not become READY within ${resource_timeout_seconds} seconds."
+    exit 1
+  fi
+
+  echo "Waiting for CatalogSource to be READY..."
+  sleep 5s
+done
+#fi
 
 
-starting_csv=$(oc get packagemanifests nutanixcsioperator -o jsonpath=\{.status.channels[*].currentCSV\})
-source=$(oc get packagemanifests nutanixcsioperator -o jsonpath=\{.status.catalogSource\})
-source_namespace=$(oc get packagemanifests nutanixcsioperator -o jsonpath=\{.status.catalogSourceNamespace\})
+# Get the package manifest for the Nutanix CSI Operator from the specific catalog
+packagemanifests_nutanix=$(oc get packagemanifests -o json | jq -r --arg catalog "$catalog" '.items[] | select(.metadata.name=="nutanixcsioperator" and .status.catalogSourceDisplayName==$catalog)')
+
+# Extract required variables
+starting_csv=$(echo "$packagemanifests_nutanix" | jq -r '.status.channels[].currentCSV')
+source=$(echo "$packagemanifests_nutanix" | jq -r '.status.catalogSource')
+source_namespace=$(echo "$packagemanifests_nutanix" | jq -r '.status.catalogSourceNamespace')
+default_channel=$(echo "$packagemanifests_nutanix" | jq -r '.status.defaultChannel')
+
+# Debugging output
+echo "Catalog: $catalog"
+echo "Starting CSV: $starting_csv"
+echo "Source: $source"
+echo "Source Namespace: $source_namespace"
+echo "Default Channel: $default_channel"
+
 
 cat <<EOF | oc apply -f -
 apiVersion: operators.coreos.com/v1alpha1
 kind: Subscription
 metadata:
+  labels:
+    operators.coreos.com/nutanixcsioperator.openshift-cluster-csi-drivers: ""
   name: nutanixcsioperator
   namespace: openshift-cluster-csi-drivers
 spec:
-  installPlanApproval: Automatic
+  channel: ${default_channel}
```

Comment:
if we use the default channel - we don't need to add the channel here
but, for debug it's maybe good idea to print it